src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/CornerRadius/CornerRadiusEffect.shared.cs

```diff
@@ -1,82 +1,47 @@
 ﻿using System;
-using System.ComponentModel;
-using System.Linq;
 using Xamarin.Forms;
 using Xamarin.Forms.Shapes;
 
 namespace Xamarin.CommunityToolkit.Effects
 {
-	public class CornerRadiusEffect : NullEffect
+	public class CornerRadiusEffect
 	{
 		public static readonly BindableProperty CornerRadiusProperty = BindableProperty.CreateAttached(
 			nameof(CornerRadius),
 			typeof(CornerRadius),
 			typeof(CornerRadiusEffect),
 			default(CornerRadius),
-			propertyChanged: TryGenerateEffect);
+			propertyChanged: OnCornerRadiusPropertyChanged);
 
-		public static CornerRadius GetCornerRadius(BindableObject? bindable)
-			=> (CornerRadius)(bindable?.GetValue(CornerRadiusProperty) ?? throw new ArgumentNullException(nameof(bindable)));
-
-		public static void SetCornerRadius(BindableObject? bindable, CornerRadius value)
-			=> bindable?.SetValue(CornerRadiusProperty, value);
-
-		static void TryGenerateEffect(BindableObject? bindable, object oldValue, object newValue)
+		static void OnCornerRadiusPropertyChanged(BindableObject bindable, object oldValue, object newValue)
 		{
-			if (bindable is not VisualElement elementView)
+			if (!(bindable is VisualElement elementView))
 				return;
 
-			var cornerRadiusEffects = elementView.Effects.OfType<CornerRadiusEffect>().ToArray();
-
-			if (GetCornerRadius(bindable) == default)
-			{
-				foreach (var cornerRadiusEffect in cornerRadiusEffects)
-					elementView.Effects.Remove(cornerRadiusEffect);
+			elementView.SizeChanged -= ElementViewSizeChanged;
+			elementView.SizeChanged += ElementViewSizeChanged;
```

Comment:
Just to point out, that we need to see if this will not cause a memory leak. I believe that doesn't, but just want to make sure to bring more eyes to it.

Reply:
Yes, we don't unsubscribe from the event as there is no detach method. Maybe use WeakEventManager?

Reply:
Not sure if we can use the WeakEventManager to subscribe to other events. I'll confirm if this can be an issue, if so I'll implement a fix during this week (;

Reply:
Yes, unfortunately the current implementation of `WeakEventManager` does not support subscription to other events.

Maybe an enhancement that could be added, like with WPF: https://docs.microsoft.com/en-us/dotnet/desktop/wpf/advanced/weak-event-patterns?view=netframeworkdesktop-4.8#using-an-existing-weak-event-manager-class

Reply:
I really like this idea. @brminnick do you think that could be add on WeakEventManager or would be better to have this as a separated internal feature?

Reply:
Interesting! @YZahringer is there an open Issue for it? If not, could you create one and I'll take a look!

Reply:
Taking [this old MS app](https://github.com/Microsoft/SmartHotel360-Mobile/blob/master/Source/SmartHotel.Clients/SmartHotel.Clients/Behaviors/ItemTappedCommandListView.cs#L20) as a reference it's ok to use this approach. I think that we're good to go